src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js

```diff
@@ -66,6 +67,19 @@ class BaseDrawerNavigator extends Component {
                     ),
                     swipeEdgeWidth: 500,
                 }}
+                screenListeners={{
+                    state: (e) => {
+                        const state = e.data.state;
+
+                        // Get the history that has drawer type to get the status.
+                        const hasDrawerHistory = _.find(state.history || [], h => h.type === 'drawer');
+
+                        // hasDrawerHistory will has undefined value if the route drawer is equal to initial route drawer.
+                        // Using the default status if hasDrawerHistory is undefined and get the status from the current route
+                        // if the value provided.
+                        App.setDefaultDrawerStatus(hasDrawerHistory ? hasDrawerHistory.status : state.default);
```

Comment:
When do we need this to run exactly? It seems like it will happen on every state change...

Reply:
The bug seems like it just affects the initial state of the drawer when the app first inits so do we need to set this any other time?

Reply:
We save the drawer status when it changes. 

For example when we are on the report screen, so the drawer has the status `closed`, and then we save the status to Onyx. When we refresh the page, it will still be showing the report screen, not the LHN.

Reply:
Ok so to summarize, this code is saving the "last state of the drawer" so that when we open the app again we can make sure to use that drawer state?

First question I have is whether it would be better to use the `useDrawerStatus()` hook to track this?

https://reactnavigation.org/docs/drawer-navigator#checking-if-the-drawer-is-open

Reply:
Maybe it can be placed inside one of our main components like `Expensify` and update the status there.

Reply:
Yeah, the hook is a good alternative, but in our cases, we can't use it on main components like `Expensify` because the component is not the child of the `Drawer.Navigator`. If we still want to use the hook, we need to add it to the screens component that uses the drawer navigation.

Reply:
Ok well wherever we can put it and it will work seems more straightforward than this `screenListeners` thing that is looking at a bunch of undocumented parameters that could change in the future?

Let's maybe pause this briefly while we figure out if a drawer status tracker is necessary.